### PR TITLE
Add precommit checking for code style

### PR DESCRIPTION
**Pull request by Bronzila**
This is currently marked as a draft, since we are not yet adhering to the style guidelines. However I still think it is an good idea to include it, so that we can at least check, that we do not add more style-errors.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,47 +1,50 @@
...
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: v0.0.273
     hooks:
-      - id: pydocstyle
-        files: dehb
-        additional_dependencies: ["toml"] # Needed to parse pyproject.toml
-
-  - repo: https://gitlab.com/pycqa/flake8
-    rev: 4.0.1
+      - id: ruff
```
Comment (eddiebergman) [inline .pre-commit-config.yaml:44]: :clinking_glasses:

File: pyproject.toml [hunk 2]
```diff
@@ -28,42 +28,160 @@ exclude_lines = [
...
+  "D103",
+  "ANN001",
+  "ANN201",
+  "FBT001",
+  "D100",
+  "PLR2004",
+  "PD901",   #  X is a bad variable name. (pandas)
+  "TCH",
 ]
+"src/amltk/__version__.py" = ["D100"]
```
Comment (eddiebergman) [inline pyproject.toml:143]: This too

**Review by Neeratyoy — COMMENTED** [review]
Guess we just merge this and iterate, right?
If so, please go ahead and merge it after Eddie's comments are accounted for!

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -1,47 +1,50 @@
...
-        files: tests
-
-      - id: black
-        name: black formatter examples
-        files: examples
-
-  - repo: https://github.com/pycqa/pydocstyle
-    rev: 6.1.1
+      - id: mypy
+        exclude: "test_comm_task"  # Pre-commit mypy hates this one, crashes on (l106)
```
[the remark this answers is not in the document]
Comment (Bronzila) [reply .pre-commit-config.yaml:32]: Shame on me🥴

**Pull request merged**